events/events_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package events
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd/protobuf/types"
+)
+
+func TestEnvelope_Field(t *testing.T) {
+	e := &Envelope{
+		Namespace: "my-namespace",
+		Topic:     "my-topic",
+		Event: &types.Any{
+			TypeUrl: "foobar",
+			Value:   []byte("hello world"),
+		},
+	}
+
+	tests := []struct {
+		name      string
+		fieldpath []string
+		want      string
+		want1     bool
+	}{
+		{
+			name:      "get namespace",
+			fieldpath: []string{"namespace"},
+			want:      "my-namespace",
+			want1:     true,
+		},
+		{
+			name:      "get topic",
+			fieldpath: []string{"topic"},
+			want:      "my-topic",
+			want1:     true,
+		},
+		{
+			name:      "get event",
+			fieldpath: []string{"event"},
+			want:      "",
+			want1:     false,
+		},
+		{
+			name:      "get topic by multi filed",
+			fieldpath: []string{"topic", "namespace"},
+			want:      "my-topic",
+			want1:     true,
+		},
+		{
+			name:      "filed path is unknown",
+			fieldpath: []string{"unknown"},
+			want:      "",
+			want1:     false,
+		},
+		{
+			name:      "filed path is null",
+			fieldpath: []string{},
+			want:      "",
+			want1:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := e.Field(tt.fieldpath)
```

Comment:
nit: `ok` is generally used as the name for a bool indicating whether a value was retrieved

```suggestion
			got, ok := e.Field(tt.fieldpath)
```